tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/FilterStaticFrameworksTaskTest.cs

```diff
@@ -0,0 +1,241 @@
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using NUnit.Framework;
+using Xamarin;
+using Xamarin.MacDev.Tasks;
+
+#nullable enable
+
+namespace Xamarin.MacDev.Tasks.Tests {
+
+	[TestFixture]
+	public class FilterStaticFrameworksTaskTest {
+
+		string tempDir = "";
+
+		[SetUp]
+		public void Setup ()
+		{
+			tempDir = Cache.CreateTemporaryDirectory ();
+		}
+
+		[TearDown]
+		public void TearDown ()
+		{
+			if (Directory.Exists (tempDir))
+				Directory.Delete (tempDir, true);
+		}
+
+		[Test]
+		public void TestCustomFrameworkExecutablePath_iOS ()
+		{
+			TestCustomFrameworkExecutablePathForPlatform ("iOS", "libavcodec.framework", "libavcodec.dylib", false);
+		}
+
+		[Test]
+		public void TestCustomFrameworkExecutablePath_tvOS ()
+		{
+			TestCustomFrameworkExecutablePathForPlatform ("tvOS", "libavcodec.framework", "libavcodec.dylib", false);
+		}
+
+		[Test]
+		public void TestCustomFrameworkExecutablePath_macOS ()
+		{
+			TestCustomFrameworkExecutablePathForPlatform ("macOS", "libavcodec.framework", "libavcodec.dylib", true);
+		}
+
+		[Test]
+		public void TestCustomFrameworkExecutablePath_MacCatalyst ()
+		{
+			TestCustomFrameworkExecutablePathForPlatform ("MacCatalyst", "libavcodec.framework", "libavcodec.dylib", true);
+		}
```

Comment:
Use separate TestCase attributes on a single test for each platform, instead of seperate test methods.

Also use the ApplePlatform enum to specify the platform, instead of hardcoding a string value.